### PR TITLE
Add a promise queue to fetch results in batches

### PR DESCRIPTION
**Pull request by abarmat**
### Motivation

We are fetching contract information from the RPC node by using a promise that sends multiple requests simultaneously. In some cases the parallel requests make the node to fail to respond due to throttling.

### Changes

- Add a promise queue to send at most 10 simultaneous request.

### CODE REVIEW

**Review by tmigone — APPROVED** [review]
Looks good!

File: package.json [hunk 1]
```diff
@@ -68,7 +68,7 @@
     "hardhat-deploy": "^0.7.0-beta.9",
     "hardhat-gas-reporter": "^1.0.1",
     "inquirer": "8.0.0",
-    "p-queue": "^6.6.2",
+    "p-queue": "6.6.2",
```
Comment (tmigone) [inline package.json:71]: What's the reason for this change?

Comment (abarmat) [reply package.json:71]: @tmigone the p-queue suffered changes after major version 6, we could still have the major anchored to 6, I'll change it
